utils/text_formatting.jsx

```diff
@@ -49,13 +49,15 @@ export function formatText(text, inputOptions) {
     }
 
     let output = text;
-
     const options = Object.assign({}, inputOptions);
+    const hasPhrases = (/"([^"]*)"/).test(options.searchTerm);
 
-    if (options.searchMatches) {
+    if (options.searchMatches && !hasPhrases) {
         options.searchPatterns = options.searchMatches.map(convertSearchTermToRegex);
     } else {
-        options.searchPatterns = parseSearchTerms(options.searchTerm).map(convertSearchTermToRegex);
+        options.searchPatterns = parseSearchTerms(options.searchTerm).map(convertSearchTermToRegex).sort((a, b) => {
+            return b.term.length - a.term.length;
+        });
```

Comment:
Here I added a sort by length to make sure the order of input search terms won't affect the highlighted result.